### ISSUE TITLE
glb model was rotated 90 degree in current master branch version

### ISSUE THREAD

**Issue by kladess** [AnalyticalGraphicsInc/cesium#6713]
Developed using glb and czml, a source for master branch is being built and used.
All 3D models were rotated CCW direction  in master branch.

using Cesium 1.46.1 version
![1_46_1](https://user-images.githubusercontent.com/6187128/41722403-6999c782-75a3-11e8-82da-0d341a77e077.png)

using 488ca6701a883923b9cecb2e858531eef1012364 version

![current_master_branch_issue](https://user-images.githubusercontent.com/6187128/41722404-69bf89ea-75a3-11e8-8b18-89c0392238aa.png)

**Comment by lilleyse** [comment]
This change was added in https://github.com/AnalyticalGraphicsInc/cesium/pull/6632 to orient glTF 2.0 models with Cesium's forward axis. But it looks like this is a case where the conversion is not desired.

@emackey do you know if there might be a clean way to control the forward axis setting from within the CZML and entity layers?

**Comment by mramato** [comment]
@lilleyse  If it's a 2.0 model, isn't this desired behavior?  Didn't we list #6632 in breaking changes?

**Comment by lilleyse** [comment]
Yes it's expected behavior, however the forward axis conversion is really intended for objects in motion https://github.com/AnalyticalGraphicsInc/cesium/pull/6632#issuecomment-392893516.

There is a gray area for buildings and other things that don't have a concept of a forward axis, which is why we disabled the rotation for 3D Tiles. This case is more similar to the 3D Tiles case, just that its czml + glb instead.

**Comment by kladess** [comment]
I attach some of the glb and czml I am using.

[output.zip](https://github.com/AnalyticalGraphicsInc/cesium/files/2125830/output.zip)

**Comment by hpinkos** [comment]
@lilleyse should we close this issue?  Can you update the comment in `CHANGES.md` to include what our recommendation is for our users with existing models with the wrong forward axis?  We need an answer here before the release

**Comment by mramato** [comment]
@emackey any thoughts here?  I'm of the opinion that this is a breaking change regarding Cesium's interpretation of glTF 2.0 model orientation and should be listed as such in CHANGES.  However, the change itself is good and I think we just need to close this issue.

If we can't come to a consensus, then we might have to back out the orientation changes until we can.

**Comment by hpinkos** [comment]
@mramato It is already listed in under `Breaking Changes`, but I think we need to add a recommendation for a path forward for existing models using an incorrect orientation to meet our previously wrong implementation.  That's why I labeled this `next release, doc`. After that I agree, this issue can be closed.  Unless I'm missing some other detail.

**Comment by mramato** [comment]
This comment from @lilleyse is what I was concerned about:

> There is a gray area for buildings and other things that don't have a concept of a forward axis, which is why we disabled the rotation for 3D Tiles. This case is more similar to the 3D Tiles case, just that its czml + glb instead.

I don't think we should special case orientation anywhere (3D Tiles doesn't count because the spec allow you to override up axis specifically).

**Comment by kladess** [comment]
i was rotated 90 degrees on the y-axis all models . But it seems that everyone needs to turn around all of the models they've built for the previous version, but they need instructions and tools.

**Comment by lilleyse** [comment]
It's unfortunate that is a breaking change for models that aren't necessarily wrong. They just don't have an implicit forward axis (for example, this model has heading and altitude baked in).

<img width="208" alt="view1" src="https://user-images.githubusercontent.com/915398/41916477-20edb328-7926-11e8-988a-5aead5651de8.PNG">
<img width="213" alt="view2" src="https://user-images.githubusercontent.com/915398/41916478-21010cf2-7926-11e8-969b-fb763d70d8a1.PNG">

The breaking changes entry might be like:

> * glTF 2.0 models corrected to face +Z forwards per specification.  Internally Cesium uses +X as forward, so a new +Z to +X rotation was added for 2.0 models only. To fix models that are oriented incorrectly after this change, apply a +X to +Z rotation `[0,0,1,0,0,1,0,0,-1,0,0,0,0,0,0,1]` to the root node of the glTF.

The fix feels a bit arbitrary and I hope not too many models need it, but it might be the best we can do.

**Comment by lilleyse** [comment]
CHANGES.md edits are here https://github.com/AnalyticalGraphicsInc/cesium/pull/6738.

@kladess since this problem affects your data, do you think the updated instructions are clear?
